packages/core/src/document.ts

```diff
@@ -378,17 +378,77 @@ class Document extends EventEmitter {
     return (await this._findIndex(cid, log)) !== -1
   }
 
+  /**
+   * Given two different DocStates representing two different conflicting histories of the same
+   * document, pick which version to accept, in accordance with our conflict resolution strategy
+   * @param state1 - first log's state
+   * @param state2 - second log's state
+   * @returns true if state2's log should be taken, or false if state1's log should be taken
+   * @private
+   */
+  static async _pickLogToAccept(state1: DocState, state2: DocState): Promise<boolean> {
+    const isState1Anchored = state1.anchorStatus === AnchorStatus.ANCHORED
+    const isState2Anchored = state2.anchorStatus === AnchorStatus.ANCHORED
+
+    if (isState1Anchored != isState2Anchored) {
+      // When one of the logs is anchored but not the other, take the one that is anchored
+      return isState2Anchored
+    }
+
+    if (isState1Anchored && isState2Anchored) {
+      // compare anchor proofs if both states are anchored
+      const { anchorProof: proof1 } = state1
+      const { anchorProof: proof2 } = state2
+
+      if (proof1.chainId == proof2.chainId) {
+        // The logs are anchored in the same blockchain, compare block heights to decide which to take
+        if (proof1.blockNumber < proof2.blockNumber) {
+          return false
+        } else if (proof2.blockNumber < proof1.blockNumber) {
+          return true
```

Comment:
```suggestion
        if (proof1.blockNumber !== proof2.blockNumber) {
          return proof2.blockNumber < proof1.blockNumber
```
Is this simpler?

Reply:
Nope, doesn't work because that would catch the case where the block numbers are equal, which is totally possible if the conflicting records were sent to different CAS instances.  Without using the fallback method of picking the lower CID, you could wind up with diverging histories in this case as which record gets chosen would depend on which record each node heard about first.

Reply:
The first if-statement I suggested explicitly checks that the block numbers are not equal, so this should not be the case. Or am I missing something here?